crates/ln-dlc-node/src/tests/bitcoind.rs

```diff
@@ -0,0 +1,56 @@
+use crate::tests;
+use anyhow::bail;
+use anyhow::Result;
+use reqwest::Response;
+use serde::Deserialize;
+use std::time::Duration;
+use tests::FAUCET_ORIGIN;
+
+#[derive(Deserialize, Debug)]
+struct BitcoindResponse {
+    result: String,
+}
+
+pub async fn fund(address: String, amount: bitcoin::Amount) -> Result<Response> {
+    query(format!(
+        r#"{{"jsonrpc": "1.0", "method": "sendtoaddress", "params": ["{}", "{}"]}}"#,
+        address,
+        amount.to_btc()
+    ))
+    .await
+}
+
+/// Instructs `bitcoind` to generate to address.
+pub async fn mine(n: u16) -> Result<()> {
+    for _ in 0..n {
+        let response =
+            query(r#"{"jsonrpc": "1.0", "method": "getnewaddress", "params": []}"#.to_string())
+                .await?;
+        let response: BitcoindResponse = response.json().await.unwrap();
+
+        query(format!(
+            r#"{{"jsonrpc": "1.0", "method": "generatetoaddress", "params": [1, "{}"]}}"#,
+            response.result
+        ))
+        .await?;
```

Comment:
What's the response of this? If it is the TX-id then we can get rid of the sleep below and return the tx ids. The caller can then do a wait-loop and check if the wallet learned about the tx-id.

Reply:
It's the block hash.

https://developer.bitcoin.org/reference/rpc/generatetoaddress.html